diagnostics/collector.go

```diff
@@ -0,0 +1,170 @@
+package diagnostics
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/treeverse/lakefs/db"
+)
+
+type Collector struct {
+	db db.Database
+}
+
+const (
+	maxRecordsPerQueryCollect = 1000
+	csvFileExt                = ".csv"
+)
+
+var ErrNoColumnsFound = errors.New("no columns found")
+
+func NewCollector(adb db.Database) *Collector {
+	return &Collector{
+		db: adb,
+	}
+}
+
+func (c *Collector) Collect(ctx context.Context, w io.Writer) (err error) {
+	writer := zip.NewWriter(w)
+	defer func() { err = writer.Close() }()
+
+	var errs []error
+	contentFromTables := []string{
+		"auth_installation_metadata",
+		"schema_migrations",
+		"pg_stat_database",
+	}
+	for _, tbl := range contentFromTables {
+		err = c.writeTableContent(ctx, writer, tbl)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("write table content for %s %w", tbl, err))
+		}
+	}
+
+	countFromTables := []string{
+		"catalog_branches",
+		"catalog_commits",
+		"catalog_repositories",
+		"auth_users",
+	}
+	for _, tbl := range countFromTables {
+		err = c.writeTableCount(ctx, writer, tbl)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("write table count for %s %w", tbl, err))
+		}
+	}
+
+	err = c.writeRawQueryContent(ctx, writer, "table_sizes", `
+SELECT *, pg_size_pretty(total_bytes) AS total
```

Comment:
@tzahij for a quick glance..

Reply:
@tzahij can you comment if you have or not have any comment related to this query?